<!--Home Page -->
<!--Designed by https://github.com/spriyansh-->

<!DOCTYPE html>
<html lang="en" dir="ltr">
  <head>
    <!-- Logos -->
    <script src="https://kit.fontawesome.com/a050a6ddf3.js" crossorigin="anonymous"></script>

    <meta charset="utf-8">
    <!-- Bootstrap Bundle -->
    <meta name="viewport" content="width=device-width, initial-scale=1">
    <link href="https://cdn.jsdelivr.net/npm/bootstrap@5.0.2/dist/css/bootstrap.min.css" rel="stylesheet" integrity="sha384-EVSTQN3/azprG1Anm3QDgpJLIm9Nao0Yz1ztcQTwFspd3yD65VohhpuuCOmLASjC" crossorigin="anonymous">

    <!-- Index CSS -->
    <link rel="stylesheet" href="css/index.css">

    <title>CSG-21</title>
  </head>
  <body>

    <!-- Division block 1 for page heading -->
    <div class="container-fluid div_block_1">
      <h1 class="fs-1 fw-bold">How inference pipelines impact Co-occurrence networks?</h1>
    </div>

    <!-- Division block 2 for background -->
<div class="container-fluid pt-0 p-3 div_block_2">
  <h3 class="">Background</h3>
  <p class="pt-1">Ever since the discovery of DADA pipeline, that generates exact sequence variants instead of clustered sequence variants, there has been a debate in the metagenomics consortium. Even though numerous studies have been published stating that there is no statistically significant difference between the two if high similarity cut-offs are used. The impact of these strategies on microbial co-occurrence networks is still unanswered. The microbes do not dwell in isolation; instead, they thrive in colonies and form associations. These associations shape the patterns and structure of their microscopic world, which administers the macroscopic world. To analyse these associative patterns, microbiologists have implemented the concepts of networks science onto these associations. The graphical form of these pairwise associations is called a microbial co-occurrence network. To draw ecologically sound inferences from these networks, incorporation of phylogenetic measures is also crucial as it increases interpretability and aids in elucidating the community assembly process.</p>
</div>

<!-- Division Block 3 for Objective -->
<div class="container-fluid pt-0 p-3 div_block_3">
  <h3 class="fs-3">Objectives</h3>
  <p class="pt-1">This project will test whether clustering microbiome gene sequences at different levels of sequence similarity (i.e. 97% to 100%) affects the inferred microbial interactions in correlation networks, particularly regarding the relative importance of habitat-linked. We will be using existing soil microbiome datasets from Chernobyl Exclusion Zone (Ukraine), in addition to public datasets.</p>
</div>

<!-- Division for exploration links -->
<div class="container-fluid div_block_explore">
      <div class="row">
        <div class="feature-box col-lg-3"><a href="pipelines.html">
          <i class="icon fas fa-cogs fa-4x"></i>
          <h3 class="fs-3 h3_v2">Pipelines</h3></a>
        </div>

        <div class="feature-box col-lg-3">
          <i class="icon fas fa-poll fa-4x"></i>
          <h3 class="fs-3 h3_v2">Results</h3>
        </div>

        <div class="feature-box col-lg-3">
          <i class="icon fas fa-database fa-4x"></i>
          <h3 class="fs-3 h3_v2">Datasets</h3>
        </div>

        <div class="feature-box col-lg-3">
          <i class="icon fas fa-exclamation-circle fa-4x"></i>
          <h3 class="fs-3 h3_v2">Misc</h3>
        </div>
      </div>
      </div>

<!-- Divison for footer -->
<footer class="text-center footer">
Footer
</footer>


  </body>
</html>
