\chapter*{Appendix}

\pagenumbering{arabic} \setcounter{page}{21}

\subsection*{DADA2 Pipeline}
The follwoing pipeline is written in R version 4.0.5 (2021-03-31)

\subsubsection*{Syntax for loading libraries}

\begin{lstlisting}[language=R]
# Required Libraries
library(dada2) # Version 1.18.0
library(ggplot2) # Version 3.3.3
library(DECIPHER) # Version 2.18.1
library(phangorn) # Version 2.5.5
library(phyloseq) # Version 1.34.0
\end{lstlisting}

\subsubsection*{Syntax for setting filepaths and conventions}

\begin{lstlisting}[language=R]
# Reading path for untrimmed fastq files
untrimmed <- file.path("fastqFiles/")

# Sorting the filenames
fns <- sort(list.files(untrimmed, full.names = TRUE))

# Roots for forward reads
fnFs <- fns[grepl("_R1", fns)]

# Roots for reverse reads
fnRs <- fns[grepl("_R2", fns)]

# Directory for filtered output (Removal of Ns)
filtFs <- file.path("filtered",fnFs) # Forward
filtRs <- file.path("filtered",fnRs) # Reverse

names(filtFs) <- fnFs # Forward
names(filtRs) <- fnRs # Reverse
\end{lstlisting}

\subsubsection*{Syntax for quality control (QC-Trim)}

\begin{lstlisting}[language=R]
out <- filterAndTrim(fwd = fnFs, # Roots for forward reads
  filt = filtFs, # Path for filtered forward reads
  rev = fnRs, # Roots for reverse reads
  filt.rev = filtRs, # Path for filtered reverse reads
  maxN = 0, # Removing Ns
  maxEE = c(2,2),
  truncQ = 0, truncLen = 245, # Trimming start/stop
  rm.phix=FALSE, multithread=TRUE,
  compress = T)
\end{lstlisting}

\subsubsection*{Syntax for dereplicating}

\begin{lstlisting}[language=R]
derepFs <- derepFastq(filtFs)
derepRs <- derepFastq(filtRs)
sam.names<-sapply(strsplit(basename(filtFs),"_"),'[', 1)
names(derepFs) <- sam.names
names(derepRs) <- sam.names
\end{lstlisting}

\subsubsection*{Syntax for error model estimation}

\begin{lstlisting}[language=R]
# Forward
ddF <- dada(derepFs[1:10], err=NULL, selfConsist=TRUE)

# Reverse
ddR <- dada(derepRs[1:10], err=NULL, selfConsist=TRUE)
\end{lstlisting}

\subsubsection*{Syntax for running DADA}

\begin{lstlisting}[language=R]
# Forward
dadaFs <- dada(derepFs, err=ddF[[1]]$err_out, pool=TRUE)

# Reverse
dadaRs <- dada(derepRs, err=ddR[[1]]$err_out, pool=TRUE)
\end{lstlisting}

\subsubsection*{Syntax for Merging and Extracting ASV}

\begin{lstlisting}[language=R]
# Merging forwards and Backward reads
mergers <- mergePairs(dadaFs, derepFs, dadaRs, derepRs)

# Extracting Seqs
seqtab.all <- makeSequenceTable(mergers)

# Removing Chimeras
seqtab <- removeBimeraDenovo(seqtab.all)
\end{lstlisting}

\subsubsection*{Syntax for Assigning Taxonomy with RDP taxonomic training data 
formatted for DADA2 (RDP trainset 16/release 11.5)}

\begin{lstlisting}[language=R]
# Using assignTaxonomy() from dada2
asv_rdp_tax <- assignTaxonomy(seqtab,
"trainingSets/rdp_train_set_16.fa.gz , multithread=TRUE)
\end{lstlisting}

\subsection*{\newline MOTHUR Pipeline}
The follwoing pipeline is from MOTHUR Version 1.35.1

\subsubsection*{Syntax for making contigs}

\begin{lstlisting}[language=R]
make.contigs(file=stability.files, processors=8)
\end{lstlisting}

\subsubsection*{Syntax for trimming}

\begin{lstlisting}[language=R]
screen.seqs(fasta,groups,maxambig=0,maxlength=275)
\end{lstlisting}

\subsubsection*{Syntax for trimming}

\begin{lstlisting}[language=R]
screen.seqs(trim.fasta,stability.groups,maxlength=275)
\end{lstlisting}

\subsubsection*{Syntax for dereplication}

\begin{lstlisting}[language=R]
unique.seqs(fasta=stability.trim.contigs.good.fasta)
\end{lstlisting}

\subsubsection*{Syntax for taxanomic Assignment}

\begin{lstlisting}[language=R]
align.seqs(fasta=unique.fasta,reference=silva.v4.fasta)
\end{lstlisting}


\subsubsection*{Syntax for OTU clustering}

\begin{lstlisting}[language=R]
pre.cluster(unique.fasta,otu_table,diffs=2)
\end{lstlisting}

\subsubsection*{Syntax for Counting OTUs}

\begin{lstlisting}[language=R]
chimera.vsearch(cluster.fq,otu_table,dereplicate=t)
\end{lstlisting}

\subsubsection*{Syntax for removing chimeras}

\begin{lstlisting}[language=R]
remove.seqs(fasta=precluster.fq,accnos=denovo.fq)
\end{lstlisting}